src/tests/ftest/cart/util/cart_logtest.py

```diff
@@ -209,7 +209,7 @@ def subtract(self, val):
                "DER_NOTLEADER(-2008): 'Not service leader'")
 
 # Functions that are never reported as errors.
-IGNORED_FUNCTIONS = ('sched_watchdog_post', 'rdb_timerd')
+IGNORED_FUNCTIONS = ('sched_watchdog_post', 'rdb_timerd', 'cont_open')
```

Comment:
Why are we ignoring cont_open here instead of fixing the errors? And what are the errors we're ignoring?

Reply:
This is some temporary workaround for the following NLT warning:
https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-14912/10/VM_test/folder.-412761578/

`cont_open() 07139793/8d9ab078/6908b833: No permission to open the container with flags 2, capas 7/0 warning in strict mode`

I do not know how the new warning is valuable for us.

Reply:
I do not know how to fix such NLT warning because even if I modified some existing logic line in `cont_open`, the CI test will also report similar NLT warning. @daltonbohning , any suggestion for that? Thanks!

Reply:
Sorry, that was the wrong code snippet. But I think the problem is the same

Reply:
> I think the problem is it doesn't follow the standard format with the rc in the error. The same would apply for other error messages as well

You mean that we need log the "rc" in the D_ERROR(), right?

Reply:
@daltonbohning , what is the new NLT warning means:

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14984/7/VM_test/new/

The warning corresponding to the following changes in the patch (https://github.com/daos-stack/daos/pull/14984):
`
2300         if (((flags & DAOS_COO_EX) && !ds_sec_cont_can_open_ex(sec_capas)) ||
2301             ((flags & DAOS_COO_RW) && !ds_sec_cont_can_modify(sec_capas))) {
2302                 rc = -DER_NO_PERM;
2303                 D_ERROR(DF_UUID "/" DF_UUID "/" DF_UUID ": failed to open the container "
2304                         "with flags " DF_X64 ", capas " DF_X64 "/" DF_X64 ": " DF_RC "\n",
2305                         DP_UUID(cont->c_svc->cs_pool_uuid), DP_UUID(pool_hdl->sph_uuid),
2306                         DP_UUID(cont->c_uuid), flags, pool_hdl->sph_sec_capas, sec_capas,
2307                         DP_RC(rc));
`

Reply:
Sorry, I'm not familiar enough with NLT to know what it's complaining about.
Looking at the code, it might be complaining that the error is logged at all?

Reply:
Honestly, I do not know. The NLT warning is still there even if without my change. But because I added more information in such log line, then it failed related CI test results. Personally, I do not think such warning is valuable for us. That is why I put `cont_open` into the ignore list.

Reply:
According to the commit edeafc14d87a5d839f4dba482efc7a54cca89592 that introduced the IGNORED_FUNCTIONS list, I did not find any special reason for adding "rdb_timerd" into such list except reducing unnecessary "fail" report.

Reply:
In fact, such NLT warning for `cont_open` log exists on current master, it is not reported as CI failure if no patch touch related code. But my patch touches it because I need more information in the log, then it failed the CI test.

From the test logic perspective (although I do not agree), it seems that such log line should not be there.
@daltonbohning , do you still think it is not an acceptable solution to "skip" the existing `cont_open` NLT warning?

Reply:
Here is an example for the NLT warning existence:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15055/1/VM_test/

Reply:
I'm not very familiar with NLT so I don't know what is best here. It just seems bad to disable a subset of testing because it's failing. @ashleypittman is the best source here but is on sabbatical.
If it's pre-existing, maybe this applies? Copied from our dev channel

>Tips from Ashley:
the way Jenkins works is it checks if messages are the same, and obviously it can't just check the line number for this as that changes.  It takes a snapshot of the 5 lines before and after the line where the error is reported, and if these are the same then the error is treated as the same.  If these lines are modified then the error does not match and therefore you get "one removed error", "one error added"
>
>in which case the procedure is the same as for adding a new error, complete your PR as you want it, run with Allow-unstable-test: true commit tags, the PR will need force-landing but once it's landed master will run and create new reference builds so other PRs should not be affected.

I.e. add `Allow-unstable-test: true` to your commit pragmas and request force landing, since it's a pre-existing error

Reply:
let me try.